components/odh-notebook-controller/controllers/notebook_controller_test.go

```diff
@@ -161,6 +162,70 @@ var _ = Describe("The Openshift Notebook controller", func() {
 
 	})
 
+	// New test case for RoleBinding reconciliation
+	When("ReconcileRoleBindings is called for a Notebook", func() {
```

Comment:
I don't understand how this test can be passing since there's nothing to set `SET_PIPELINE_RBAC` to `true`. For this testing the controller is not running from an openshift deployments, so setting the variable to true in the yamls can't be doing the trick, no?

Reply:
uh, nevermind, you commented on that already

Reply:
>Need to update the tests, in case that found SET_PIPELINE_RBAC=false to don't run the tests

That's not great. It means we end up running the test exactly never, since this var is gonna be always false. So the newly added code is not covered by (running) tests.

Reply:
This is related to the new env that introduce today. 
If the  `SET_PIPELINE_RBAC` is true then it should run the tests. if fakse it should skip them, since the controller actually don't cretae rolebindings

Reply:
and when you do `make test`, it will run with SET_PIPELINE_RBAC=true first, and than SET_PIPELINE_RBAC=false next?

Reply:
Ok, i added two times the test as following:

```
.PHONY: test
test: test-with-rbac-false test-with-rbac-true
```

I can not thing any other better way to test it in both situations

Reply:
I actually like it, running the whole suite with each setting. I think this is what we should be doing. Maybe it could be done more nicely, but I also don't really see how. When we have more than one variable like this, we'd have to come up with a smarter way. For now, fine with me.

Reply:
Ty, personally, I would prefer the way that the suite can identify what is the value of the params.env file and decide if will the test cases or not.

like: 

```
test: manifests generate fmt vet envtest ## Run tests.
	export $$(cat config/manager/params.env | xargs) && \
	ACK_GINKGO_DEPRECATIONS=1.16.5 \
	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" \
		go test ./controllers/... -ginkgo.v -ginkgo.progress -test.v -coverprofile cover.out
```
If the `SET_PIPELINE_RBAC is true` on the `params.env ` then run the test and if it is false just skip them with 
```
		BeforeEach(func() {
			// Skip the tests if SET_PIPELINE_RBAC is not set to "true"
			fmt.Printf("SET_PIPELINE_RBAC is: %s\n", os.Getenv("SET_PIPELINE_RBAC"))
			if os.Getenv("SET_PIPELINE_RBAC") != "true" {
				Skip("Skipping RoleBinding reconciliation tests as SET_PIPELINE_RBAC is not set to 'true'")
			}
		})
```

Imagine in the future to have more cases like - we will have to run the full suite x times.
Probably we can create an issue to investigate that will proper time.